src/display_context.cpp

```diff
@@ -87,21 +87,47 @@ display_context::can_move_result display_context::unit_can_move(const unit& u) c
 	const team& current_team = get_team(u.side());
 
 	can_move_result result = {false, false};
-	for(const map_location& adj : get_adjacent_tiles(u.get_location())) {
-		if (map().on_board(adj)) {
-			if(!result.attack_here) {
-				const unit_map::const_iterator i = units().find(adj);
-				if (i.valid() && !i->incapacitated() && current_team.is_enemy(i->side())) {
-					result.attack_here = true;
+	if(u.attacks_left() > 0 && !u.attacks().empty()) {
+		const auto& attacks = u.attacks();
+
+		std::set<int> attackable_distances;
+	    for (const auto& attack : attacks) {
+	        for (int i = attack.min_range(); i <= attack.max_range(); ++i) {
+	            attackable_distances.insert(i);
```

Comment:
This should probably have an `if(attack.attacks_used() <= u.attacks_left())`, but I'd be happy to merge as-is and add that afterwards (it would need testing with the UtBS units, and I haven't done that yet).

If pushing another commit to this, the coding standard is to have no space between `for` or `if` and their opening round bracket, even though we keep the space before the brace at the end of the line.

Reply:
I check if attacks_left>0 same as the mainline code does, so I didn't want to differ...
Should I put check for every weapon therecinstead?

Reply:
It needs a double-check of exactly how weapons with `attacks_used() > 1` act; I think I know, but would need to spend a while testing to confirm. If you want to do that then it would be nice, but alternatively I'm happy to merge now.

Reply:
I have no experience with that... only place I know is Utbs.
I generally like the idea of early merge, adding more features later.
...
So you can be happy while merging the PR :D.